drpcconn/conn.go

```diff
@@ -66,17 +67,29 @@ func (c *Conn) Invoke(ctx context.Context, rpc string, in, out drpc.Message) (er
 	mon.Event("outgoing_requests")
 	mon.Event("outgoing_invokes")
 
-	data, err := proto.Marshal(in)
-	if err != nil {
-		return errs.Wrap(err)
-	}
-
 	stream, err := c.man.NewClientStream(ctx)
 	if err != nil {
 		return err
 	}
 	defer func() { err = errs.Combine(err, stream.Close()) }()
 
+	invokeMetadata := make([]byte, 0)
+	metadata, ok := drpcmetadata.Get(ctx)
+	if ok {
+		invokeMetadata, err = metadata.Encode(invokeMetadata)
+		if err != nil {
+			return err
+		}
+		if err := stream.RawWrite(drpcwire.KindInvokeMetadata, invokeMetadata); err != nil {
+			return err
+		}
+	}
```

Comment:
I'd prefer if we kept all of the `stream.RawWrite` calls inside of `doInvoke`. So maybe we can change this to:

```go
var metadata []byte
if md, ok := drpcmetadata.Get(ctx); ok {
    metadata, err = md.Encode(nil)
    if err != nil {
        return err
    }
}

stream, err := c.man.NewClientStream(ctx)
if err != nil {
...
}
defer ...

if err := c.doInvoke(stream, []byte(rpc), data, metadata, out); err != nil {
...
```